src/worker/hwmonitor.rs

```diff
@@ -1,21 +1,37 @@
 use serde::{Deserialize, Serialize};
-use systemstat::{saturating_sub_bytes, Platform, System};
+use std::thread;
+use std::time::Duration;
+use systemstat::{BTreeMap, CPULoad, Memory, Network, Platform, System};
 
-#[derive(Deserialize, Serialize, Debug)]
+#[derive(Deserialize, Serialize, Debug, Default)]
 pub struct WorkerHwState {
-    pub worker_cpu_usage: i8,
-    pub worker_memory_usage: i8,
-    pub worker_network_usage: i8,
+    pub worker_cpu_usage: Option<CPULoad>,
+    pub worker_memory_usage: Option<Memory>,
+    pub worker_network_usage: Option<BTreeMap<String, Network>>,
 }
 
-//fixme: what should I do with read error? read error shouldn't make it to the average
-// should a read error lead to a none in the hwState optional and a warning?
-pub fn get_worker_hw_state() -> Option<WorkerHwState> {
+pub fn get_worker_hw_state() -> WorkerHwState {
     let sys = System::new();
+    let mut hw_state: WorkerHwState = Default::default();
+    match sys.networks() {
+        Ok(networks) => hw_state.worker_network_usage = Some(networks),
+        Err(x) => {
+            log::warn!("\nNetwork usage read error: {}", x)
```

Comment:
I don't expect this to happen, but does this warning make any sense?

Reply:
It should be logged in some way, this is OK.

However, the network state should be sent in some aggregated way. Use https://docs.rs/systemstat/0.1.8/systemstat/platform/common/trait.Platform.html#tymethod.network_stats instead of `networks()` please.